scripts/pysa_fuzzer/run2.py

```diff
@@ -0,0 +1,187 @@
+import subprocess
+import os
+import json
+import argparse
+from pathlib import Path
+import shutil
+import logging
+import random
+from code_generator2 import CodeGenerator
+
+logging.basicConfig(level=logging.INFO)
+
+def run_command(command, output_file=None):
+    try:
+        result = subprocess.run(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True, check=True)
+        if output_file:
+            with open(output_file, 'w') as file:
+                file.write(result.stdout)
+    except subprocess.CalledProcessError as e:
+        logging.error(f"Command '{command}' failed with error: {e.stderr}")
+
+def generate_random_seed(num_files, num_mutations=48):
+    return [random.randint(1, num_mutations) for _ in range(num_files)]
+
+def apply_mutation(generator, mutation_number):
+    if 1 <= mutation_number <= 24:
+        # Apply source mutation
+        mutation_method = getattr(generator, f"source_mutation_{mutation_number}")
+        mutation_method()
+    elif 25 <= mutation_number <= 48:
+        # Apply sink mutation
+        mutation_method = getattr(generator, f"sink_mutation_{mutation_number - 24}")
+        mutation_method()
```

Comment:
could we move this in the CodeGenerator? Otherwise we are leaking some implementation details of the code generator. I would expect the API to be `CodeGenerator.apply_mutation(random_integer)`.

And since we are doing some introspection uglyness, we can actually find automatically all mutations so we don't have to hardcode the 24 and 48:
`source_mutations = {int(name[len('source_mutation_'):]) for name in dir(CodeGenerator) if name.startswith('source_mutation_')}`